### DISCUSSION TITLE
Detailed Product Permissions

### DISCUSSION THREAD

**Discussion by andris-sevcenko** [craftcms/commerce#2294] [Ideas] [answered]
> Created by: Alex Collins ([alex@ethercreative.co.uk](alex@ethercreative.co.uk)) on 2017/04/26 06:16:54 +0000
Votes at time of UserVoice import: 3

------
Give product type user permissions the same granularity as a Channel in CMS

**Comment by andris-sevcenko** [comment]
> Created by: Maarten ([wiejeben@gmail.com](wiejeben@gmail.com)) on 2017/09/28 08:05:14 +0000


------
I would also like to be able to let users disable the creation of products, however they may edit existing ones.

**Comment by andris-sevcenko** [comment]
> Created by: Alan Cantes ([alan@t10interactive.co.uk](alan@t10interactive.co.uk)) on 2017/12/10 14:46:35 +0000


------
+1 to this. It&#39;d be great for a client to have a user group for product editors, allowing them admin access to create products but not publish them.

**Comment by andris-sevcenko** [comment]
> Created by: Anonymous ([seskie007@gmail.com](seskie007@gmail.com)) on 2018/02/19 06:57:08 +0000


------
This feature is very important, i wanted to give user groups access to create products in commerce, and when they log in , each group type should only see their products

**Comment by andrelopez** [comment]
This feature will be perfect to add support to Craft Commerce for Stripe Connect with our Stripe Payments plugin. This will help to easily associate a vendor with a product on Marketplaces.

https://github.com/enupal/stripe/issues/1

**Comment by lukeholder** [comment]
@andrelopez Do you need an author/owner field on a product? Right now products have no author like entries.
Are you wanting only the creator of the product to edit/delete it?

**Comment by andrelopez** [comment]
Hi @lukeholder Yes exactly, I need a way to associate a Product to a User (Vendor), we currently have `Vendors` elements which has associated a  User. We also have `Connects` elements where you associate one or more products to one vendor. Basically a Vendor could have one or more products. One way is to have the author feature like entries, the other way that I was thinking is to add a filter to the ProductsQuery if a Vendor is logged in the CP where I can join with Connects to only show the products that are created by that Vendor. Please let me know if that makes sense

The `beforePrepare` function of the `ElementQuery` class rise a `CancelableEvent` but this only has the `isValid` property,  it would be great If I can have the query as a property to modify the query on a similiar event.

**Comment by lukeholder** [comment]
- [ ] For each product type gets a new edit, create and delete permissions.
- [ ] Migration for users permissions table.
- [ ] Project config user group migration

**Comment by lukeholder** [comment, marked as answer]
Will be fixed by https://github.com/craftcms/commerce/pull/1905
